lib/application/personal_information/personal_information_bloc.dart

```diff
@@ -0,0 +1,25 @@
+// ignore_for_file: invalid_use_of_visible_for_testing_member
+
+import 'package:bloc/bloc.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:injectable/injectable.dart';
+
+import '../../domain/profile/user_profile.dart';
+import '../../domain/user/i_profile_repository.dart';
+
+part 'personal_information_event.dart';
+part 'personal_information_state.dart';
+part 'personal_information_bloc.freezed.dart';
+
+@injectable
+class PersonalInformationBloc
+    extends Bloc<PersonalInformationEvent, PersonalInformationState> {
+  final IProfileRepository _profileRepository;
+  PersonalInformationBloc(this._profileRepository) : super(_Fetching()) {
+    _profileRepository.getUserProfile().then((profileOrFailure) {
+      profileOrFailure.fold((l) {}, (profile) {
```

Comment:
You should handle the error state as well.

Reply:
I'm not sure how we should deal with this error, since you shouldn't be able to reach the Account Information screen without being logged in, and thus having a profile.

What do you think is best @wizlif ?

I see these few options:
- Create an anonymous version of the Widget that uses the profile (We'll use that as an error widget)
- Show loading state (We'll implement a shimmer version), and a snackbar with an error